cmd/gardener-scheduler/app/app.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+	goruntime "runtime"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/component-base/version"
+	"k8s.io/component-base/version/verflag"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardenerhealthz "github.com/gardener/gardener/pkg/healthz"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/scheduler/apis/config"
+	"github.com/gardener/gardener/pkg/scheduler/controller"
+	schedulerfeatures "github.com/gardener/gardener/pkg/scheduler/features"
+	"github.com/gardener/gardener/pkg/server/routes"
+)
+
+// Name is a const for the name of this component.
+const Name = "gardener-scheduler"
+
+// NewCommand creates a new cobra.Command for running gardener-scheduler.
+func NewCommand() *cobra.Command {
+	opts := &options{}
+
+	cmd := &cobra.Command{
+		Use:   Name,
+		Short: "Launch the " + Name,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			verflag.PrintAndExitIfRequested()
+
+			if err := opts.complete(); err != nil {
+				return err
+			}
+			if err := opts.validate(); err != nil {
+				return err
+			}
+
+			log, err := logger.NewZapLogger(opts.config.LogLevel, opts.config.LogFormat)
+			if err != nil {
+				return fmt.Errorf("error instantiating zap logger: %w", err)
+			}
+
+			logf.SetLogger(log)
+			klog.SetLogger(log)
```

Comment:
It might be also used by underlying packages (client-go, controller-runtime), so just setting it here doesn't harm and ensures all logs follow the same style.